### ISSUE TITLE
Kontainer-engine creates serviceaccount in default namespace

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#17725]
This issue affects all versions from v2.0.0 - v2.0.10 and v2.1.0 - v2.1.5. It allows any project member with access to the default namespace to use a service account created by the Rancher control plane. This service account can be used to execute administrator level API calls against the Kubernetes cluster.

The name of the service account is “netes-default”. It is created in the “default” namespace and given the “cluster-admin” clusterRole. This grants full administrator privileges to the cluster. To understand this vulnerability, you must understand that once a service account is created in a namespace, anyone with member level privileges in that namespace is allowed to use that service account. This is known and expected service account behavior (see [the Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) for more details). However, the vulnerability arises from the fact that Rancher placed the service account in the default namespace and that namespace into the Default project. This namespace and project are expected to be used by non-admin users to create and manage workloads. The service account should not be exposed in a normal member-level namespace or project. It should be created in a namespace and project that only administrators have access to.

You can see the offending items here:
```
$ kubectl get sa -o yaml -n default
apiVersion: v1
items:
- apiVersion: v1
 kind: ServiceAccount
 metadata:
   creationTimestamp: "2018-11-30T09:24:44Z"
   name: default
   namespace: default
   resourceVersion: "265"
   selfLink: /api/v1/namespaces/default/serviceaccounts/default
   uid: c6495312-f481-11e8-9001-fa163e3f8425
 secrets:
 - name: default-token-2q2x9
- apiVersion: v1
 kind: ServiceAccount
 metadata:
   creationTimestamp: "2018-11-30T09:25:21Z"
   name: netes-default
   namespace: default
   resourceVersion: "602"
   selfLink: /api/v1/namespaces/default/serviceaccounts/netes-default
   uid: dc468abf-f481-11e8-8c42-fa163e1c7076
 secrets:
 - name: netes-default-token-pbzjx
kind: List
metadata:
 resourceVersion: ""
 selfLink: ""

$ kubectl get clusterrolebindings -n default | grep -i default
netes-default-clusterRoleBinding                       14d
```

**Comment by cjellick** [comment]
Fixed in v2.1.6 and v2.0.11

**Comment by cjellick** [comment]
# Rancher Priviledge Escalation Mitigation Instructions
## Introduction
This document will walk you through the steps to mitigate the [CVE-2018-20321](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-20321) bug on versions 2.0.0 - 2.0.10 and 2.1.0 - 2.1.5 of Rancher.  This bug works by allowing any project member with access to the `default` namespace to mount the `netes-default` service account in a pod and then use that pod to execute administrative privileged commands against the k8s cluster.  These mitigation instructions isolate the `default` namespace in a separate project where only cluster admins can be given permissions to access. This bug currently affects ALL clusters created by Rancher (but not clusters merely Imported by Rancher) and the following procedures should be followed on each cluster created by Rancher.

## Mitigation Steps for Rancher 2.0.7 and later
1. Verify the `netes-default` service account is present in your affected cluster.  Go to the cluster screen in Rancher and click `Launch kubectl`.  When the command prompt appears, type `kubectl get serviceaccounts`. You should see something like the following:
```
> kubectl get serviceaccounts
NAME          SECRETS   AGE
default       1         4m
netes-default 1         4m
```
2. Find the `default` namespace, it is under the `Default` project by default but may be in a different project if it was moved.
3. Identify all of the workloads deployed in the `default` namespace and migrate those workloads to a different namespace, creating a new namespace if necessary. By migrate we mean you must delete the workload from the default namespace and recreate it in a different namespace.
4. Click the `...` next to the `default` name space and select `Move`. Select the `System` project and click `Move`.  The `default` namespace will now be located in the `System` project.
5. Assure that no users are assigned to the `System` project. This will prevent users from escalating their priviledges with the `netes-default` service account. 
6. Repeat the steps 1-5 for each affected cluster.

## Mitigation Steps for Rancher 2.0.0 - 2.0.6
1. Verify the `netes-default` service account is present in your affected cluster.  Go to the cluster screen in Rancher and click `Launch kubectl`.  When the command prompt appears, type `kubectl get serviceaccounts`. You should see something like the following:
```
> kubectl get serviceaccounts
NAME          SECRETS   AGE
default       1         4m
netes-default 1         4m
```
2. Navigate to the `Projects/Namespaces` page and click `Add Project`. Enter `System Default` in the name field and click `Ok` to create the project. Note that we have chosen the name "System Default" for this project because in Rancher v2.1.x, we automatically create a project named "System" and we want to avoid confusion. You may choose any name you like for this project.
3. Find the `default` namespace, it is under the `Default` project by default but may be in a different project if it was moved.
4. Identify all of the workloads deployed in the `default` namespace and migrate those workloads to a different namespace, creating a new namespace if necessary. By migrate we mean you must delete the workload from the default namespace and recreate it in a different namespace.
5. Click the `...` next to the `default` name space and select `Move`. Select the `System Default` project and click `Move`.  The `default` namespace will now be located in the `System Default` project.
6. Assure that no users are assigned to the `System Default` project. This will prevent users from escalating their privileges with the `netes-default` service account. 
7. Repeat the steps 1-6 for each affected cluster.
